src/home/selectors.ts

```diff
@@ -31,69 +28,5 @@ export const getExtraNotifications = createSelector(
   }
 )
 
-export const cleverTapInboxMessagesSelector = createSelector(
-  (state: RootState) => state.home.cleverTapInboxMessages,
-  (rawMessages: ExpectedCleverTapInboxMessage[]) => {
-    const CLEVERTAP_INBOX_TAG = 'CleverTapInbox'
-
-    const messages: CleverTapInboxMessage[] = []
-
-    if (!Array.isArray(rawMessages)) {
-      Logger.error(CLEVERTAP_INBOX_TAG, 'Unexpected CleverTap Inbox messages format', {
-        rawMessages,
-      })
-
-      return messages
-    }
-
-    for (const rawMessage of rawMessages) {
-      const messageId = rawMessage.id ?? rawMessage._id
-      const content = rawMessage.msg?.content?.[0]
-      const header = content?.title?.text
-      const text = content?.message?.text
-      const iconUrl = content?.icon?.url
-      const icon = iconUrl ? { uri: iconUrl } : undefined
-      const action = content?.action?.links?.[0]
-      const ctaText = action?.text
-      const ctaUrl =
-        Platform.OS === 'android'
-          ? action?.url?.android?.text
-          : Platform.OS === 'ios'
-          ? action?.url?.ios?.text
-          : ''
-
-      const PRIORITY_TAG = 'priority:'
-      const priority = Number(
-        rawMessage.msg?.tags
-          ?.find((tag: string) => tag.startsWith(PRIORITY_TAG))
-          ?.slice(PRIORITY_TAG.length)
-      )
-
-      const openInExternalBrowser = rawMessage.msg?.tags?.includes('openInExternalBrowser') ?? false
-
-      if (!messageId || !text || !ctaText || !ctaUrl) {
-        Logger.error(CLEVERTAP_INBOX_TAG, 'Unexpected CleverTap Inbox message format', {
-          messageId,
-          text,
-          ctaText,
-          ctaUrl,
-          rawMessage,
-        })
-        continue
-      }
-
-      messages.push({
-        messageId,
-        header,
-        text,
-        icon,
-        ctaText,
-        ctaUrl,
-        priority: !Number.isNaN(priority) ? priority : undefined,
-        openInExternalBrowser,
-      })
-    }
-
-    return messages
-  }
-)
+export const cleverTapInboxMessagesSelector = (state: RootState) =>
```

Comment:
should we add the feature flag here too, and return empty if not enabled?

Reply:
totally! thanks for the suggestion. i also added a gate inside the event emitter.